lib/tasks/contacts.rake

```diff
@@ -37,4 +37,67 @@ namespace :contacts do
       ::Contacts.publishing_api.put_content_item(contact.link, p.present)
     end
   end
+
+  desc "Remove a contact and redirect it"
+  task :remove_with_redirect, [:contact_slug, :redirect_to_location] => :environment do |_task, args|
+    if args.contact_slug.blank? || args.redirect_to_location.blank?
+      puts "Usage: rake contacts:remove_with_redirect[contact-to-remove-slug,path-to-redirect-to]"
+    else
+      contact = Contact.find_by_slug(args.contact_slug)
+      if contact.nil?
+        puts "Contact #{args.contact_slug} doesn't exist, consider using rake contacts:replace_gone_with_redirect"
+      else
+        print "Contact #{contact.link} removed with redirect to #{args.redirect_to_location} "
+        if Admin::DestroyAndRedirectContact.new(contact, args.redirect_to_location).destroy_and_redirect
+          puts "SUCCESS"
+        else
+          puts "ERROR"
+        end
+      end
+    end
+  end
+
+  desc "Put in place a redirect for an already removed contact"
+  task :replace_gone_with_redirect, [:contact_slug, :organisation_slug, :redirect_to_location] => :environment do |_task, args|
```

Comment:
This task includes some significant conditional logic which isn't tested - if this work were done by a separate object then we could test that the arguments are used correctly and that we're only creating the redirect under the right conditions.